File: java/leetcode-practice/src/test/java/com/mamta/leetcode/Leet11.java
package com.mamta.leetcode;

import org.junit.jupiter.api.Test;

import static java.lang.Math.min;

public class Leet11 {

    public int maxArea(int[] height) {
        int left = 0;
        int right = height.length-1;
        int max = 0;

        while(left<right) {
            int width = right - left;
            int area = min(height[right] , height[left]) * width;
            max = Math.max(area, max);

            if(height[left] < height[right]) {
                left++;
            } else {
                right--;
            }
        }
        return max;
    }

    @Test
    public void solution() {
        int[] height = {1,8,6,2,5,4,8,3,7};
        int capacity = maxArea(height);

        System.out.println("Capacity: " + capacity);
    }
}

File: python/two_pointer/leet11.py
from typing import List


class Solution:
    def maxArea(self, height: List[int]) -> int:
        left = 0
        right = len(height)-1
        max_water = 0
        while left < right:
            width = right-left
            area = min(height[left], height[right])*width
            max_water = max(area, max_water)

            if height[left] < height[right]:
                left += 1
            else:
                right-=1
        return max_water


if __name__ == "__main__":
    s = Solution()
    height = [1, 8, 6, 2, 5, 4, 8, 3, 7]
    print(s.maxArea(height))
    # Area Formula -> a =